docs: Add API and CLI documentation (#3917)
dagger/dagger: docs/current/api/990050-index.md

---
slug: /api
---

# Dagger GraphQL API

## What is the Dagger GraphQL API?

Dagger exposes its functionality through a GraphQL API. GraphQL serves as a low-level language-agnostic API framework for all Dagger operations.

> In practice, interacting with the API directly is optional. You can instead use a Dagger SDK for one of the supported languages, or use the Dagger CLI.

Reviewer: ```suggestion
In practice, interacting with the API directly is optional. We encourage users to use instead a Dagger SDK for one of the supported languages, or use the Dagger CLI.
```

> 

Reviewer 2: A simple stack diagram here would be useful.

Reviewer: https://github.com/mermaid-js/mermaid/issues/3307  :eyes:

---
slug: /api
---

> # Dagger API

Author: I want to understand this terminology better in our specific context.
- Should this be "Dagger Engine" instead?
- What's the difference between the API and the Engine?
- Is the CLI an interface to the API, to the Engine, or both?
